docs/pages/2020_BIGMAP/sections/basics.rst

```diff
@@ -3,3 +3,658 @@
+    This is because different entities, such as nodes, computers and groups, are stored in different tables of the database.
+    So, the PKs for each entity type are unique for each database, but entities of different types can have the same PK within one database.
+
+Let's now start up the ``verdi shell`` again and load the ``add@tutor`` code using its label:
+
+.. code-block:: ipython
+
+    In [1]: code = load_code(label='add')
+
+Every code has a convenient tool for setting up the required input, called the *builder*.
+It can be obtained by using the ``get_builder`` method:
+
+.. code-block:: ipython
+
+    In [2]: builder = code.get_builder()
+
+Using the builder, you can easily set up the calculation by directly providing the input arguments.
+Let's use the ``Int`` node that was created by our previous ``calcfunction`` as one of the inputs and a new node as the second input:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(pk=4)
+       ...: builder.y = Int(5)
+
+In case that your nodes' PKs are different and you don't remember the PK of the output node from the previous calculation, check the provenance graph you generated earlier and use the UUID of the output node instead:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(uuid='42541d38')
+       ...: builder.y = Int(5)
+
+Note that you don't have to provide the entire UUID to load the node.
+As long as the first part of the UUID is unique within your database, AiiDA will find the node you are looking for.
+
+.. note::
+
+    One nifty feature of the builder is the ability to use tab completion for the inputs.
+    Try it out by typing ``builder.`` + ``<TAB>`` in the verdi shell.
+
+To execute the ``CalcJob``, we use the ``run`` function provided by the AiiDA engine:
+
+.. code-block:: ipython
+
+    In [4]: from aiida.engine import run
+       ...: run(builder)
+
+Wait for the process to complete.
+Once it is done, it will return a dictionary with the output nodes:
+
+.. code-block:: ipython
+
+    Out[4]:
+    {'sum': <Int: uuid: 7d5d781e-8f17-498a-b3d5-dbbd3488b935 (pk: 8) value: 11>,
+    'remote_folder': <RemoteData: uuid: 888d654a-65fb-4da0-b3bc-d63f0374f274 (pk: 9)>,
+    'retrieved': <FolderData: uuid: 4733aa78-2e2f-4aeb-8e09-c5cfb58553db (pk: 10s)>}
+
+Besides the sum of the two ``Int`` nodes, the calculation function also returns two other outputs: one of type ``RemoteData`` and one of type ``FolderData``.
+See the :ref:`topics section on calculation jobs <topics:calculations:usage:calcfunctions>` for more details.
+Now, exit the IPython shell and once more check for *all* processes:
+
+.. code-block:: console
+
+    $ verdi process list -a
+    PK  Created    Process label             Process State    Process status
+    ----  ---------  ------------------------  ---------------  ----------------
+    185  39m ago    multiply                  ⏹ Finished [0]
+    189  1m ago     ArithmeticAddCalculation  ⏹ Finished [0]
```

Comment:
These pk seem to be inconsistent with what was being used before (the first `multiply` had pk of 3). I'm not sure if this is the first instance of this, I think it is, but it is anyways repeated afterwards (see example below).